install/install.sh

```diff
@@ -49,6 +49,35 @@ complete() {
   printf "${GREEN}✓${NO_COLOR} $@\n"
 }
 
+# Gets path to a temporary file, even if 
+get_tmpfile(){
+  local suffix
+  suffix="$1"
+  if hash mktemp; then
+    printf "$(mktemp).${suffix}"
+  else
+    # No really good options here--let's pick a default + hope
+    printf "/tmp/starship.%s" "${suffix}"
+  fi
+}
+
+# Test if a location is writeable by trying to write to it. Windows does not let
+# you test writeability other than by writing: https://stackoverflow.com/q/1999988
+test_writeable(){
+  local path
+  path="${1:-}"
+  set +e
+  touch "${path}/test.txt" 2> /dev/null
+  rc=$?
+  set -e
+  if [ $rc -eq 0 ]; then
+    rm "${path}/test.txt"
```

Comment:
Huh. Did not know that putting commands in a conditional appeases `set -e`. Thanks!

Reply:
@andytom Are you sure about that one?

I made a dummy example and it seems that toggling `-e` is still required. 🤔 
```bash
#!/bin/bash

set -euo pipefail

fn() {
  echo "start"
  if true; then
    false # simulate failing command
    printf "result: %s" $?
  fi
  echo "end" # this never gets printed!
}

fn
```

Reply:
I think the problem you are seeing is because that script is testing if the `true` command  passed not the `false` command, the `if` just checks the exit code of the command/function that was run, see below.

```bash
#!/bin/bash
set -euo pipefail

fn() {
  echo "start"
  if /bin/true; then # a passing command
    echo "This should always get printed"
  fi
  if /bin/false; then # a failing command
    echo "This should never get printed"
  fi
  echo "end"
}

fn
```

Reply:
@andytom Of course, I was thinking that we are talking about commands inside blocks instead of conditions which is totally wrong so just ignore me 🤦 Anyway, thank you for the nice explanation! 👍